src/main/java/com/zufar/onlinestore/cart/converter/ShoppingSessionDtoConverter.java

```diff
@@ -1,29 +1,35 @@
 package com.zufar.onlinestore.cart.converter;
 
+import com.zufar.onlinestore.cart.api.ItemsTotalPriceCalculator;
 import com.zufar.onlinestore.cart.dto.ShoppingSessionDto;
 import com.zufar.onlinestore.cart.dto.ShoppingSessionItemDto;
 import com.zufar.onlinestore.cart.entity.ShoppingSession;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 
+import java.math.BigDecimal;
 import java.util.Collection;
 
 @Component
 @RequiredArgsConstructor
 public class ShoppingSessionDtoConverter {
 
     private final ShoppingSessionItemDtoConverter shoppingSessionItemDtoConverter;
+    private final ItemsTotalPriceCalculator itemsTotalPriceCalculator;
 
     public ShoppingSessionDto toDto(final ShoppingSession entity) {
         Collection<ShoppingSessionItemDto> items = entity.getItems().stream()
```

Comment:
so maybe try to use list here directly to avoid ambiguities

Reply:
Done